index.js

```diff
@@ -67,6 +81,31 @@ module.exports = function (src, dest, opts) {
 				resolve(true);
 			});
 
+			var progressEnabled = progress.emitter.listeners('progress').length > 0;
+			if (progressEnabled) {
+				read.on('data', function () {
+					var written = write.bytesWritten;
+					var percent = written / progress.stat.size;
+					progress.emitter.emit('progress', {
+						src: src,
+						dest: dest,
+						size: progress.stat.size,
+						written: written,
+						percent: percent
+					});
+				});
+
+				read.on('end', function () {
```

Comment:
> I tried, but sometimes it does not gets emitted.

From [`fs.createWriteStream`](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options):

> If autoClose is set to true (default behavior) on error or end the file descriptor will be closed automatically.

But, I trust in your assessment.